app/ts/TradingUI/Trading.tsx

```diff
@@ -58,6 +57,21 @@ const TradingView = ({ maybeReadClient, maybeWriteClient, marketData, currentTim
 		}
 	})
 
+	const updateShareBalances = async () => {
+		if (maybeWriteClient.deepValue === undefined) return
+		if (marketData.deepValue === undefined) return
+		const shareBalances = await getShareBalances(maybeWriteClient.deepValue, marketData.deepValue.marketAddress, maybeWriteClient.deepValue.account.address)
+		invalidBalance.deepValue = shareBalances[0]
+		noBalance.deepValue = shareBalances[1]
+		yesBalance.deepValue = shareBalances[2]
+	}
+
+	useSignalEffect(() => {
+		if (maybeWriteClient.deepValue === undefined) return
+		if (marketData.deepValue === undefined) return
+		updateShareBalances()
+	})
```

Comment:
`updateShareBalances` will early return immediately already for both of these conditions, so no need for them, which lets you remove the entire wrapping function and just have `updateShareBalances` as your effect functions directly.
```suggestion
	useSignalEffect(updateShareBalances)
```

That being said, if `getShareBalances` throws you'll get an `unhandled promise rejection` in the browser so you should handle that.  Perhaps something like:
```suggestion
	useSignalEffect(() => updateShareBalances().catch(console.error)
```